bakery/mgorootkeystore/rootkey.go

```diff
@@ -132,9 +177,68 @@ func (b backing) FindLatestKey(createdAfter, expiresAfter, expiresBefore time.Ti
 	return key, nil
 }
 
+// InsertKey implements dbrootkeystore.Backing.
 func (b backing) InsertKey(key dbrootkeystore.RootKey) error {
-	if err := b.coll.Insert(key); err != nil {
+	return insertKey(b.coll, key)
+}
+
+// InsertKeyContext implements dbrootkeystore.ContextBacking.
+func (b backing) InsertKeyContext(ctx context.Context, key dbrootkeystore.RootKey) error {
+	var err error
+
+	f := func(coll *mgo.Collection) {
+		err = insertKey(coll, key)
+	}
+
+	if err := b.runWithContext(ctx, f); err != nil {
+		return err
+	}
+
+	return err
+}
+
+func insertKey(coll *mgo.Collection, key dbrootkeystore.RootKey) error {
+	if err := coll.Insert(key); err != nil {
 		return errgo.Notef(err, "mongo insert failed")
 	}
 	return nil
 }
+
+func (b backing) runWithContext(ctx context.Context, f func(*mgo.Collection)) error {
+	s := sessionFromContext(ctx)
+	if s == nil {
+		s = b.coll.Database.Session
+	}
+	s = s.Clone()
+
+	c := make(chan struct{})
+
+	go func() {
```

Comment:
is this run in a go routine just so we can check if the context is cancelled in the meantime? sure, runWithContext will return once the context is cancelled.. but the go routine will still be running.. do we care about that?